### PR TITLE
Improve docs on instance building functions

### PR DESCRIPTION
**Pull request by matmarczak**
Add description with example about module level instance building functions.

Solves issue https://github.com/FactoryBoy/factory_boy/issues/572

### CODE REVIEW

**Comment by matmarczak** [comment]
Thanks for the reply. Sorry but I didn't attach right ticket to my PR. The issue I was trying to solve was https://github.com/FactoryBoy/factory_boy/issues/572. I edited my post to match what I meant.

**Comment by francoisfreitag** [comment]
Instead of an example, what do you think of guarding against use of a Factory as the `klass`? Something like:
```diff
diff --git a/factory/helpers.py b/factory/helpers.py
index 01866ec..42214f4 100644
--- a/factory/helpers.py
+++ b/factory/helpers.py
@@ -30,6 +30,9 @@ def make_factory(klass, **kwargs):
     """Create a new, simple factory for the given class."""
     factory_name = '%sFactory' % klass.__name__
 
+    if isinstance(klass, base.FactoryMetaClass):
+        raise TypeError(f"{klass!r} should not be a Factory")
+
     class Meta:
         model = klass
 
```

**Comment by matmarczak** [comment]
This solves case when one creates factory with `make_factory` and similar. I can't see case when one want to use Factory as Meta.model, so what do you think about more general approach like:

diff --git a/factory/base.py b/factory/base.py
index da7aa55..d35e1dd 100644
--- a/factory/base.py
+++ b/factory/base.py
@@ -162,8 +162,17 @@ class FactoryOptions(object):
         Custom FactoryOptions classes should override this method
         to update() its return value.
         """
+
+        def _is_factory(meta, value):
+            if isinstance(value, FactoryMetaClass):
+                raise TypeError(
+                    "%s is already a %s"
+                    % (repr(value), Factory.__name__)
+                )
+
         return [
-            OptionDefault('model', None, inherit=True),
+            OptionDefault('model', None, inherit=True,
+                          checker=_is_factory),
             OptionDefault('abstract', False, inherit=False),

I agree, that example in instance building docs section wouldn't be useful.

**Comment by francoisfreitag** [comment]
I like that suggestion!

File: docs/reference.rst [hunk 1]
```diff
@@ -2018,13 +2018,21 @@ Instance building
...
-extracting instances from them:
+extracting instances from them. Methods can be used to create factories in
+dynamic way based on parameters.
+
+Internally methods use :func:`make_factory` to create new :class:`Factory` and then
+perform additional call on newly created class according to method name.
+
+Please note, that all Factories created with this methods inherit from
+:class:`factory.base.Factory` class. For full support of your ``ORM`` you should specify
+base class with ``FACTORY_CLASS`` parameter as showed in :func:`make_factory` examples.
```
Comment (francoisfreitag) [inline docs/reference.rst]:
```suggestion
base class with ``FACTORY_CLASS`` parameter as shown in :func:`make_factory` examples.
```

Show is a passive.

**Comment by francoisfreitag** [comment]
Thanks for the patch @matmarczak!

I pushed a commit for last minute cosmetic edits to your branch. The commit message will be used for the squashed commits. Please feel free to suggest rewording or improvements. Otherwise, I’ll squash and merge in the next few days.

**Comment by francoisfreitag** [comment]
Add a last commit™ for a last cosmetic edit.
